rest_framework/middleware.py

```diff
@@ -0,0 +1,26 @@
+class BaseMiddleware:
```

Comment:
what about extending django middleware mechanism instead?

Reply:
You mean the whole flow of django middleware or just subclassing the `django.utils.deprecation.MiddlewareMixin`?

Reply:
DRF is an extension to django so we should be able to write custom middleware for DRF using django base middleware mechanism and custom it aligning with DRF's flow of reqquest response